pkg/resource/clickpipe_cdc_infrastructure.go

```diff
@@ -0,0 +1,275 @@
+			)
+			return
+		}
+		r.client = client
+	}
+}
+
+func (r *ClickPipeCdcInfrastructureResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan models.ClickPipeCdcInfrastructureModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	serviceID := plan.ServiceID.ValueString()
+
+	// Validate CPU is a multiple of 1000
+	cpuMillicores := plan.ReplicaCpuMillicores.ValueInt64()
+	if cpuMillicores%1000 != 0 {
+		resp.Diagnostics.AddError(
+			"Invalid Configuration",
+			fmt.Sprintf("replica_cpu_millicores must be a multiple of 1000, got %d", cpuMillicores),
+		)
+		return
+	}
+
+	// Validate memory is a multiple of 4
+	memoryGb := plan.ReplicaMemoryGb.ValueFloat64()
+	if int(memoryGb)%4 != 0 {
+		resp.Diagnostics.AddError(
+			"Invalid Configuration",
+			fmt.Sprintf("replica_memory_gb must be a multiple of 4, got %.1f", memoryGb),
+		)
+		return
+	}
+
+	// Validate memory is 4x CPU cores
+	cpuCores := float64(cpuMillicores) / 1000.0
+	expectedMemory := cpuCores * 4
+	if memoryGb != expectedMemory {
+		resp.Diagnostics.AddError(
+			"Invalid Configuration",
+			fmt.Sprintf("replica_memory_gb must be 4× the CPU core count. With %d millicores (%.1f cores), memory must be %.1f GB, but got %.1f GB",
+				cpuMillicores, cpuCores, expectedMemory, memoryGb),
+		)
+		return
+	}
+
+	// Poll for CDC scaling endpoint to become available (up to 10 minutes)
+	maxWait := 10 * time.Minute
+	checkAvailable := func() error {
+		_, err := r.client.GetClickPipeCdcScaling(ctx, serviceID)
+		return err
+	}
+
+	exponentialBackoff := backoff.NewExponentialBackOff(
+		backoff.WithMaxElapsedTime(maxWait),
+		backoff.WithMaxInterval(30*time.Second),
+	)
+
+	if err := backoff.Retry(checkAvailable, exponentialBackoff); err != nil {
+		resp.Diagnostics.AddError(
+			"CDC Infrastructure Not Available",
+			"The CDC infrastructure endpoint is not yet available after waiting 10 minutes. "+
+				"This endpoint becomes available once at least one DB ClickPipe has been provisioned. "+
+				"Please create a Postgres CDC ClickPipe first, then try creating this resource again.\n\n"+
+				fmt.Sprintf("Error: %s", err.Error()),
+		)
+		return
+	}
+
+	// Update the scaling settings
+	scalingReq := api.ClickPipeCdcScalingRequest{
+		ReplicaCpuMillicores: plan.ReplicaCpuMillicores.ValueInt64(),
+		ReplicaMemoryGb:      plan.ReplicaMemoryGb.ValueFloat64(),
+	}
+
+	scaling, err := r.client.UpdateClickPipeCdcScaling(ctx, serviceID, scalingReq)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Creating CDC Infrastructure",
+			fmt.Sprintf("Could not create CDC infrastructure: %s", err.Error()),
+		)
+		return
+	}
+
+	// Update plan with response values
+	plan.ReplicaCpuMillicores = types.Int64Value(scaling.ReplicaCpuMillicores)
+	plan.ReplicaMemoryGb = types.Float64Value(scaling.ReplicaMemoryGb)
```

Comment:
After UpdateCDCScaling returns, the values are "accepted", not "applied" - to properly update the plan we should wait till GetCDCScaling starts returning them (average 2-5 min)

Reply:
Makes sense, I've switched now to adding a polling function after creation and update that checks for the values. Oddly though it seems to happen much faster than you describe (circa ~15 seconds). Might be worth looking into.